pyannote/database/__init__.py

```diff
@@ -29,54 +29,74 @@
 
 """pyannote.database"""
 
+import os
+from pathlib import Path
 import sys
 from pkg_resources import iter_entry_points
 
-from typing import Optional, Dict, Set, Text
+from typing import List, Optional, Dict, Set, Text, Union
+
+from pyannote.database.registry import registry, OverrideType
 
 from .database import Database
-from .database import PyannoteDatabaseException
 
 from .protocol.protocol import Protocol
 from .protocol.protocol import ProtocolFile
 from .protocol.protocol import Subset
 from .protocol.protocol import Preprocessors
 
-from .util import FileFinder
+from .file_finder import FileFinder
 from .util import get_annotated
 from .util import get_unique_identifier
 from .util import get_label_identifier
 
 from ._version import get_versions
 
-DATABASES = dict()
-TASKS: Dict[Text, Set[Text]] = dict()
-
-from .custom import add_custom_protocols
 
 __version__ = get_versions()["version"]
 del get_versions
 
 
+# TODO : deprecate and delete, or make it work with Registry
 # load databases from entry points
-for o in iter_entry_points(group="pyannote.database.databases", name=None):
+# for o in iter_entry_points(group="pyannote.database.databases", name=None):
 
-    database_name = o.name
+#     database_name = o.name
 
-    DatabaseClass = o.load()
-    DATABASES[database_name] = DatabaseClass
+#     DatabaseClass = o.load()
+#     DATABASES[database_name] = DatabaseClass
 
-    database = DatabaseClass()
+#     database = DatabaseClass()
 
-    for task in database.get_tasks():
-        if task not in TASKS:
-            TASKS[task] = set()
-        TASKS[task].add(database_name)
+#     for task in database.get_tasks():
+#         if task not in TASKS:
+#             TASKS[task] = set()
+#         TASKS[task].add(database_name)
 
-    setattr(sys.modules[__name__], database_name, DatabaseClass)
+#     setattr(sys.modules[__name__], database_name, DatabaseClass)
 
 # parse pyannote.database configuration file, looking for custom protocols
-DATABASES, TASKS = add_custom_protocols()
+# DATABASES, TASKS = add_custom_protocols()
+
+
+def env_config_paths() -> List[Path]:
+    valid_paths = []
+
+    env_config_paths = os.environ.get("PYANNOTE_DATABASE_CONFIG", "")
+    splitted = env_config_paths.split(":")
+    for path in splitted:
+        path = Path(path).expanduser()
+        if path.is_file():
+            valid_paths.append(path)
+    return valid_paths
+
+
+# load all databases contained in the PYANNOTE_DATABASE_CONFIG env variable
+registry.load_databases(*env_config_paths())
+
+
+def load_database_yml(*paths: list, allow_override=OverrideType.WARN_OVERRIDES):
```

Comment:
This is the function that our end-user will use, right? 
Why not ask them to use the registry directly?

```python
from pyannote.database import registry
registry.load_database("/path/to/database.yml", allow_override=...)
```

instead of

```python
from pyannote.database import load_database_yml
load_database_yml("/path/to/database.yml", allow_override=...)
```

This way, we can put everything as method of this `registry` thing.

Reply:
For backward compatibility and because `get_protocol` is used in almost every tutorial, I would still expose `get_protocol` 

```python
get_protocol = registry.get_protocol
```

Users would progressively switch from `get_protocol` to `registry.get_protocol`.